Correciones 16/05/2022

agustinsa8 opened the pull request:
Se realizan las actividades propuestas por el instrutor:
1. Endpoint para crear moneda. el simbolo no puede estar repetido, el simbolo debe ser de 3 caracteres (solo los datos basicos, simbololo, nombre, icono(url))
2. Endpoint para listar todas las monedas con paginacion. usar @pageable. Cabe anexar que no se pudo implementar lo de "pageable" porque aun no se entiende el tema.

agustinsa8 committed:
Asignacion 13-05-2022 por el instructor

Se realizan las actividades propuestas por el instrutor:
1. Endpoint para crear moneda. el simbolo no puede estar repetido, el simbolo debe ser de 3 caracteres (solo los datos basicos, simbololo, nombre, icono(url))
2. Endpoint para listar todas las monedas con paginacion. usar @pageable. Cabe anexar que no se pudo implementar lo de "pageable" porque aun no se entiende el tema.

agustinsa8 committed:
Asignacion 13-05-2022 por el instructor

Se realizan las actividades propuestas por el instrutor:
1. Endpoint para crear moneda. el simbolo no puede estar repetido, el simbolo debe ser de 3 caracteres (solo los datos basicos, simbololo, nombre, icono(url))
2. Endpoint para listar todas las monedas con paginacion. usar @pageable. Cabe anexar que no se pudo implementar lo de "pageable" porque aun no se entiende el tema.

caferrerbeam commented on src/main/java/com/co/indra/coinmarketcap/coins/services/CoinService.java:
```diff
@@ -1,11 +1,37 @@
     private CoinRepository coinRepository;
+
+    public void registerCoin(String idSymbolCoin, String nameCoin, String iconCoin) {
+        if(coinRepository.findCoinByIdSymbolCoin(idSymbolCoin).size() == 0) {
```
```
if (coinRepository.findCoinByIdSymbolCoin(idSymbolCoin).isEmpty()) {
   throw new BusinessException(ErrorCodes.ID_SYMBOLCOIN_ALREDY_EXIST);
}

coinRepository.create(new Coin(idSymbolCoin, nameCoin, iconCoin));
```

agustinsa8 committed:
Correcciones del 16/05/2022

Se realizaron las correcciones correspondientes y la aplicación de las pruebas unitarias para crearCoin y para listarCoins.
